### PR TITLE
Migrate `pallet-transaction-payment` to new pallet attribute macro

### PR DESCRIPTION
**Pull request by koushiro**
Part of #7882

Migrate the `pallet-transaction-payment` to the new pallet attribute macro.

## :warning:  Breaking Change :warning:

From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

> storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the pallet must be careful about the name they used in `construct_runtime!`. Hence the `runtime-migration` label, which might not be needed depending on the configuration of the pallet.

cc @shawntabrizi @thiolliere 

`polkadot` and `kusama` use `TransactionPayment` as pallet name, thus no need for migration

polkadot companion: https://github.com/paritytech/polkadot/pull/3267

### CODE REVIEW

**Comment by thiolliere** [comment]
## ⚠️ Breaking Change ⚠️ 

From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

> storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the pallet must be careful about the name they used in `construct_runtime!`. Hence the `runtime-migration` label, which might not be needed depending on the configuration of the pallet.

File: frame/transaction-payment/src/lib.rs [hunk 1]
```diff
@@ -229,44 +230,84 @@ impl Default for Releases {
...
+		Multiplier,
+		ValueQuery,
+		NextFeeMultiplierOnEmpty
+	>;
+
+	#[pallet::storage]
+	pub(super) type StorageVersion<T: Config> = StorageValue<_, Releases, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T> {
```
Comment (thiolliere) [inline frame/transaction-payment/src/lib.rs]: I removed the generic here as it is not needed

(but also I think we should remove the GenesisConfig completly and remove the `Releases` storage. If people wonder about which version their pallet is, then they should look at the pallet version stored.
Note that substrate node sand polkadot/kusama nodes don't run this genesis config (it is not configured in construct_runtime).
But we can do in a follow-up PR)

**Comment by thiolliere** [comment]
some tests needs fix

**Review by KiChjang — APPROVED** [review]
Mostly syntax changes; if it compiles, it works.

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**